ngraph/core/src/util.cpp

```diff
@@ -155,6 +155,22 @@ size_t ngraph::hash_combine(const std::vector<size_t>& list)
     return seed;
 }
 
+size_t ngraph::hash_combine(const void* v, int64_t size)
```

Comment:
Why cannot we use the same algorithm like here:
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/inference_engine/compilation_context.cpp#L34

Reply:
This algorithm is better suited for this specific purpose, for starters we're using data's size in bytes for the seed to minimize collision chance. 

I can additionally tell you, that I tested that algorithm also, and some performance measurements I made for random models that are being used (I did this by adding IRSerialization tests that used specific models). Then I compared 
Using algorithm you proposed, I've seen the execution time go up ~10-15% for each model. However with the current approach, we're looking at ~1-2% time increase (obviously those numbers aren't super precise since I've done this on my regular dev setup)- significantly quicker.